Update the ORA notebook to use murine leukemia data

jaclyn-taroni opened the pull request:
Per our chat in Slack this morning, I am updating the ORA notebook to use murine leukemia data.  In `pathway-analysis/setup/03-leukemia_DGE.Rmd`, I perform two comparisons: high stem cell capacity cell population vs. low stem cell capacity cell population and high stem cell capacity vs. unsorted cells. These DGE results are then used in the ORA notebook.

During instruction, we'll identify the genes with higher expression in high capacity cells compared to low capacity that don't show up in the high capacity vs. unsorted comparison.

Questions/concerns for reviewers:

- Is the explanation of ☝️ in the instruction notebook appropriate/sufficient?
- How do we feel about the filtering to remove `NA` values and, subsequently, how the background set gets generated?
- Is taking the union of genes for ID conversion and then the left join unnecessarily confusing?

The data has not been fully set up yet (e.g., the DGE results are not in `/shared` yet), but wanted to get other 👀 on it.  After running `link-data.sh`, running `pathway-analysis/setup/03-leukemia_DGE.Rmd` will put the DGE results in the correct place if you'd like to test the ORA notebook out.

jaclyn-taroni commented on scripts/link-data.sh:
```diff
@@ -52,6 +57,7 @@ link_locs=(
   scRNA-seq/data/tabula-muris/fastq
   scRNA-seq/data/tabula-muris/normalized/TM_normalized.rds
   machine-learning/data/open-pbta
+  pathway-analysis/data/leukemia
```
This has been added to `/shared`

jaclyn-taroni replied:
Note to self: These files need to be updated in `/shared` pending PR approval.

jashapiro commented on pathway-analysis/01-overrepresentation_analysis.Rmd:
````diff
@@ -107,60 +116,89 @@ In this example, we will use canonical pathways which are ([ref](https://www.gse
+
+Some of these rows have `NA` values in `padj`, which [can happen for a number of reasons](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#pvaluesNA) including when all samples have zero counts.
+
+Let's filter to rows that _do not have any `NA`_ using a function called `complete.cases()`.
+
+```{r complete_cases}
+# Remove rows that are not complete (e.g., contain NAs) by filtering to only 
+# complete rows
+vs_low_df <- vs_low_df %>%
+  # The dot here means "all"
+  dplyr::filter(complete.cases(.))
````
We could switch here to `tidyr::drop_na()`, which avoids some confusion around dots.

jashapiro commented on pathway-analysis/01-overrepresentation_analysis.Rmd:
````diff
@@ -260,11 +334,21 @@ Sometimes folks consider genes from the entire genome to comprise the background
 
-```{r detected_genes, live = TRUE}
-detected_genes <- unique(as.character(neuron_markers_df$gene_symbol))
+If we are unable to include a gene because, for example, it had low mean expression in our experiment and therefore was filtered out in our `complete.cases()` step, we shouldn't include in our background set.  
````
Two comments: One is just a note that if you take my `tidyr::drop_na()` suggestion, this will need to change. The other is that you don't explicitly say (I don't think) that low mean expression is going to be filtered during that step... you do mention zero expression, which is not quite the same.

jashapiro commented on pathway-analysis/01-overrepresentation_analysis.Rmd:
```diff
@@ -107,60 +116,89 @@ In this example, we will use canonical pathways which are ([ref](https://www.gse
+In both tables of DGE results, we use Ensembl gene identifiers. 
+So we will need to convert from these identifiers into either gene symbols or Entrez IDs.
 
 ### Gene identifier conversion
```
If we move this before the filtering, we can save some steps (identical gene lists) , and also perhaps make the logic a bit more clear about why we need to do the conversion... no steps between getting the database info and making the identifiers match.

Also, you might move the above paragraphs justifying the conversion under this header?

jashapiro reviewed (commented):
Looks good. I have a few fairly minor comments, one of which is about the mechanics of dropping NA values, but the _idea_ seems good to me. It is a good demonstration of the point of background sets.

The union/left join stuff is okay, but maybe a bit confusing. It might be easier to get the identifiers before filtering, in which case the gene lists should be the same?

jaclyn-taroni commented on pathway-analysis/setup/03-leukemia_DGE.Rmd:
```diff
@@ -0,0 +1,112 @@
+                                               "MacLowKitposSca1pos", 
+                                               "unsorted")) %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column(var = "Gene") %>%
```
I want to make sure we cover gene identifier conversion during instruction. If only so I can make the joke about 70% of my PhD being spent on gene identifier conversion.

jaclyn-taroni commented:
Thanks for the review, those were helpful comments! I've rearranged things and I *believe* I've addressed all of your points. One additional change I made: I decided to move the filtering & gene ID conversion for **one** of the comparisons to the prep notebook. I also chose to make those steps a little bit different than the steps we take in the instruction notebook and to point that out so curious parties can take a look if they'd like. Let me know what you think.

jashapiro commented on pathway-analysis/01-overrepresentation_analysis.Rmd:
````diff
@@ -230,28 +263,54 @@ fisher.test(gene_table, alternative = "greater")
+```
+
+We want genes that are in the first comparison but not in the second!
+We can use `setdiff()`, a base R function for set operations, to get the list that we want.
````
Probably not something people will think much about, but I just had to look to satisfy myself that order mattered for `setdiff()`

jaclyn-taroni replied:
The comment in the code block itself is a reminder for me to say that during training. Observation: I believe you are of the opinion that people are more likely to read the text outside the comments than the comments themslves and I have the opposite opinion. My opinion is founded pretty much entirely in my own behavior/tendencies.

jashapiro commented on pathway-analysis/01-overrepresentation_analysis.Rmd:
````diff
@@ -260,11 +319,21 @@ Sometimes folks consider genes from the entire genome to comprise the background
 
-```{r detected_genes, live = TRUE}
-detected_genes <- unique(as.character(neuron_markers_df$gene_symbol))
+If we are unable to include a gene because, for example, it had low mean expression in our experiment and therefore was filtered out in our `tidyr::drop_na()` step, we shouldn't include in our background set.  
````
```suggestion
If we were unable to include a gene in one of our differential expression comparisons because, for example, it had low mean expression in our experiment and therefore was filtered out in our `tidyr::drop_na()` step, we shouldn't include in our background set.  
```

jashapiro commented on pathway-analysis/01-overrepresentation_analysis.Rmd:
```diff
@@ -260,11 +319,21 @@ Sometimes folks consider genes from the entire genome to comprise the background
-detected_genes <- unique(as.character(neuron_markers_df$gene_symbol))
+If we are unable to include a gene because, for example, it had low mean expression in our experiment and therefore was filtered out in our `tidyr::drop_na()` step, we shouldn't include in our background set.  
+
+We can use another function for set operations, `intersect()`, to get our background set.
```
I thought this could be a bit more explicit about what `intersect` is doing. It is in the comment, but do people read those?
```suggestion
We can use another function for set operations, `intersect()`, to get our background set of genes that were included in both comparisons.
```

jashapiro reviewed (approved):
LGTM! Less repetition is good. Just some tiny comments
